### DISCUSSION TITLE
Can anyone help me out in importing my custom coaxial octocopter in to  airsim ???

### DISCUSSION THREAD

**Discussion by airsimdevdd** [microsoft/AirSim#3426] [Support Q&A]
i am able to import my customized coaxial octacopter model into the blocks environment,but i am unaware of what has to be changed in the physics part for the drone so i require a proper assistance to complete my sitl

**Comment by airsimdevdd** [comment]
@ahmed-elsaharti  Hi sir we sorted out the problem of combining into single file and also i have changed the multi-rotor params  and rotor like I initialized the coaxial octocopter in multi-rotorparams.hpp and builded the air-sim with no errors  further i need your assistance in where we should we give other 4 downward props and their  rotation

**Reply by ahmed-elsaharti** [reply]
@airsimdevdd, do you need help with animating the 4 downwards rotors in UE4 or with integrating them and their control signals into the physics of the simulation?

**Reply by airsimdevdd** [reply]
@ahmed-elsaharti  yes sir and i also want the exact event graph of the drone also, I  am unable to figure it out accurately

**Reply by ahmed-elsaharti** [reply]
Animating the 4 downwards rotors in UE4 can be as easy as following the [rendering section here](https://github.com/microsoft/AirSim/wiki/hexacopter#rendering) once you've figured out how to add those extra 4 rotors to the frame in [MultirotorParams](https://github.com/microsoft/AirSim/blob/master/AirLib/include/vehicles/multirotor/MultiRotorParams.hpp). As for the control signals and the physics, I honestly have no clue how a coaxial octocopter would be controlled or simulated.

**Reply by airsimdevdd** [reply]
@ahmed-elsaharti  Sir we done that rendering section we also Updated multirotors params and rotor params,but i dont why the drone crashes into ground sometimes & prop 0 stop rotating and sometimes it is stable. i thought problem is with model and params but i changed everything to default and tried with bp_flyingspawn the same happens default drone also behaves oddly and crashes into ground when we click on play

**Reply by ahmed-elsaharti** [reply]
Would it be possible to post a screenshot or a video of the problem? I'm wondering if its related to #3415

**Reply by airsimdevdd** [reply]
@ahmed-elsaharti  No sir issue not related ,I just armed  it falls into ground up and down then comes into rest then there prop0 doesn't  rotate...!  .I have attached a image where u can see default bp flying pawn is just  set to arm and collision counts on left 
![Screenshot from 2021-03-03 10-40-11](https://user-images.githubusercontent.com/77828768/109765019-45625c00-7c1a-11eb-9034-678ce5064102.png)

**Reply by ahmed-elsaharti** [reply]
@airsimdevdd I think a gif or a video would be better illustrators of the problem but from what I understood it looks like the modifications you made to the physics could be causing this.

[Here's](https://gifyu.com/image/ZaMX) what the default behavior should be like with PX4 SITL in blocks.

**Comment by airsimdevdd** [comment]
@ahmed-elsaharti  Sir  i finally spawned the coaxial octocopter thank u so much for the help .now i need to connect to px4 for sitl with my custom drone to air-sim  .what should be changes i should make in settings.json ,i have changed some lines but it doesn't connect to px4 shows error accepting tcp-socket  binding failed .I'm stuck at this point after customizing Drone can i open the custom drone in air-sim binaries blocks or should open in unreal editor (blocks editor) please help with this one sitl with our custom drone thank  you in advance ..

**Reply by ahmed-elsaharti** [reply]
Generally these are the default settings you would need to use PX4 SITL (as stated in [the docs](https://microsoft.github.io/AirSim/px4_sitl/)):
<details>
<summary>settings.json</summary>
<br>
<pre>
{
    "SettingsVersion": 1.2,
    "SimMode": "Multirotor",
    "Vehicles": {
        "PX4": {
            "VehicleType": "PX4Multirotor",
            "UseSerial": false,
            "UseTcp": true,
            "TcpPort": 4560,
            "ControlPort": 14580,
            "Parameters": {
                "NAV_RCL_ACT": 0,
                "NAV_DLL_ACT": 0,
                "LPE_LAT": 47.641468,
                "LPE_LON": -122.140165,
                "COM_OBL_ACT": 1
            }
        }
    }
}
</pre>
</details>

However, I'm not entirely sure how you've modified your frame.
For example, by default AirSim loads the quadrotor frame and initializes 4 rotors. In PX4 SITL, the default frame being used is the 3DR iris which is a quadcopter.

If we want to use a hexacopter for example, adding `"Model": "Hexacopter",` under `VehicleType` makes AirSim load the hexacopter frame in [MultiRotorParams](https://github.com/microsoft/AirSim/blob/master/AirLib/include/vehicles/multirotor/MultiRotorParams.hpp). However, it does not change the model being used in PX4. To do that you would have to add a custom SITL frame that uses the Hexacopter mixer. I'm planning on submitting a PR with some new docs showing how to do this for a hexacopter).

In general, I would suggest trying to get PX4 SITL to work with a quadrotor first before trying to modify the drone.
To get this to work, you'll need to start SITL then start AirSim with the settings shown above.

**Comment by airsimdevdd** [comment]
![Screenshot from 2021-03-03 18-27-55](https://user-images.githubusercontent.com/77828768/109869873-6d3fd700-7c8f-11eb-9078-1b95187d10d8.png)

**Comment by airsimdevdd** [comment]
@ahmed-elsaharti  why this default quadrotor my BP_FlyingPawn is behaving odd and colliding just playing it with normal settings u can see it in the video 

https://user-images.githubusercontent.com/77828768/109948315-35bd4300-7d00-11eb-818a-bbec654a7160.mp4

**Reply by ahmed-elsaharti** [reply]
Did you modify the code at all?

**Reply by airsimdevdd** [reply]
I changed  nothing that are default settings till yesterday it was working fine but today it behaves like that same happens with my custom drone too if u click on play multiples times some times it will be stable some time it behaves like this

**Reply by airsimdevdd** [reply]
@ahmed-elsaharti  I'm struck at  this need help

**Reply by ahmed-elsaharti** [reply]
In this case i would generally recommend re-cloning and rebuilding airsim to make sure everything's unchanged.
If you end up doing this, you'll want to clone #3449 rather than master since the master branch wont compile on linux due to a bug.

Edit: the PR has been merged, cloning master should be fine now
